components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/DefaultHttpClientConnector.java

```diff
@@ -189,10 +207,23 @@ public void onSuccess(String protocol, ChannelFuture channelFuture) {
                         LOG.debug("Created the connection to address: {}",
                                   route.toString() + " " + "Original Channel ID is : " + channelFuture.channel().id());
                     }
-                    if (srcHandler != null) {
+
+                    if (protocol.equalsIgnoreCase(Constants.HTTP_SCHEME) && h1SourceHandler != null) {
```

Comment:
Shall we switch constant and the variable here to avoid possible NPEs??
```suggestion
                    if ((Constants.HTTP_SCHEME).equalsIgnoreCase(protocol) && h1SourceHandler != null) {
```